### PR TITLE
TDEVOPS-2780 | CI Changes to consume gvm

### PR DESCRIPTION
**Pull request by sundeeptessell**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

Dependent Services #
- [ ] Governance
- [ ] IAM
- [ ] Deployer
- [ ] DB Systems
- [ ] Tenant
- [ ] Software Library
- [ ] Backup
- [ ] Conductor
- [ ] Monitoring
- [ ] DMM

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Environment**:

- [ ] Dev
- [ ] Staging

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/workflows/master_pr.yml [hunk 1]
```diff
@@ -28,11 +28,15 @@ jobs:
         id: runner-type
         run: |
           runner=$(yq '.runner' convoy.yaml)
+          if [[ $runner == "amd-go-22" ]]; then
+            echo "Overriding the runner from ${runner} to amd"
+            runner="amd"
```
Comment (varenyatessell) [inline .github/workflows/master_pr.yml:33]: Why only for `amd-go-22` ?

File: .github/workflows/master_build_pr.yml [hunk 2]
```diff
@@ -545,7 +547,7 @@ jobs:
     if: ${{ inputs.type == 'amd-go-22' }}
     runs-on: amd-go-22
     env:
-      CI_BRANCH: main
+      CI_BRANCH: TDEVOPS-2780-1
```
Comment (varenyatessell) [inline .github/workflows/master_build_pr.yml:550]: We can now remove every go specific builds altogether keeping only `build-amd`

File: .github/workflows/master_pr.yml (hunk 1 above)
Comment (sundeeptessell) [reply .github/workflows/master_pr.yml:33]: Yes, need to include all. I used this for testing.

File: .github/workflows/master_build_pr.yml (hunk 2 above)
Comment (sundeeptessell) [reply .github/workflows/master_build_pr.yml:550]: Yes.

**Pull request merged**